user-guide/with-ingress-gce.md

```diff
@@ -0,0 +1,163 @@
+# Install with Google Kubernetes Engine (GKE) Ingress 
+
+Google offers a [L7 load balancer](https://cloud.google.com/kubernetes-engine/docs/concepts/ingress) to 
+leverage network services like managed SSL certificates, SSL offloading or the Google content delivery network. 
+A L7 load balancer in front of Ambassador can be configured by hand or by using the ingress-gce resource. Using the 
+ingress resource also allows you to create google managed SSL certificates through kubernetes.
+
```

Comment:
So in the default setup, you do Google HTTPS Load Balancer --> ingress-gce --> Pods, and with this setup, you're swapping out the ingress-gce for Ambassador. Is that correct? Do you think you could add a note to this effect, and perhaps a reason why you wouldn't use ingress-gce? (I've read the Google docs on this, and found them to be a bit confusing)

Reply:
No. I still keep ingress-gce. Since Ingress GCE is no real service or deployment it will just be a way to configure the load balancer. So the ingress-gce will do the configuration and creation of the Google HTTPS/HTTP LB. It will configure it to point to the services listed in the ingress spec. (backend and frontend configuration) It will also install SSL certificates on the load balancer. Since there will be only one service (ambassador) it is in fact LB -> Ambassador. You could also add services that are connected directly to the LB by adding the servivces to the ingress spec. 

Do you think I need to clarify this?

Reply:
Got it. I  think it's really helpful to clarify. I'm pretty familiar with Kubernetes, and even I got confused ;-). 

Just for my understanding, if you are using ingress-gce and no Ambassador, how does the LB connect to your Kubernetes services? Do they have be exposed as NodePorts?

Reply:
I've changed the description that will hopefully clarify the setup.

The services have to be exposed as NodePort since the backend service will just be configured as instance groups plus a static port (the NodePort). That is also true for other services exposed via the ingress-gce.

There is a "new" way to set it up, that is by using Network Endpoint Groups (NEGs). I'm testing this setup soon and would update the documentation then. The ingress-gce will then create Network Endpoint Groups and route the traffic directly to the pods instead of going via NodePort and then needing an extra network hop.